### ISSUE TITLE
Better Exception handling for database errors

### ISSUE THREAD

**Issue by amot-dev** [amot-dev/mototwist#53]
Specifically, Twist name length should be passed along to the user. Perhaps there are other "common" errors that should have human-friendly messages too.
